jdaviz/core/template_mixin.py

```diff
@@ -2248,9 +2316,34 @@ def add_item(self, item):
         ----------
         item : QTable, QTableRow, or dictionary of row-name, value pairs
         """
-        def json_safe(item):
+        def json_safe(column, item):
+            def float_precision(column, item):
+                if column in ('slice', 'index'):
+                    # stored in astropy table as a float so we can also store nans,
+                    # but should display in the UI without any decimals
+                    return int(item)
+                elif column in ('pixel'):
+                    return f"{item:0.03f}"
+                else:
+                    return f"{item:0.05f}"
+
+            if isinstance(item, SkyCoord):
+                return item.to_string('hmsdms', precision=4)
+            if isinstance(item, u.Quantity) and not np.isnan(item):
+                return (float_precision(column, item.value) * item.unit).to_string()
```

Comment:
Maybe irrelevant but I wonder if you can take advantage of https://github.com/astropy/astropy/blob/c0a24c1dc957a3b565294213f435fefb2ec99714/astropy/utils/misc.py#L385 for some of the stuff here. 💭

Reply:
we probably could use our own encoder, but I don't think we want the same behavior as here so I'm not sure it's worth it.